src/peripheral/dwt.rs

```diff
@@ -370,53 +401,94 @@ pub enum ComparatorFunction {
 pub enum DwtError {
     /// Invalid combination of [AccessType] and [EmitOption].
     InvalidFunction,
+    /// The DWT block does not implement cycle count capabilities.
+    NoCycleCount,
 }
 
 impl Comparator {
     /// Configure the function of the comparator
     #[allow(clippy::missing_inline_in_public_items)]
-    pub fn configure(&self, settings: ComparatorFunction) -> Result<(), DwtError> {
+    pub fn configure(&self, dwt: &DWT, settings: ComparatorFunction) -> Result<(), DwtError> {
```

Comment:
I misread the standard. I thought `self` could tell us if this comparator supported CYCCNT, but that information is indeed only in `DWT_NOCYCCNT`. Standard says
> Comparator 0 supports cycle count comparisons only if the DWT_CTRL.NOCYCCNT bit is RAZ

but we have no way to check if this is comparator zero; with the new check CYCCNT can still be configured for a comparator that isn't zero.

We'll have to compromise: revert the new check and add a note in the documentation.

Reply:
I don't think it's the worst thing in the world to **also** check if the HW is even capable of `CYCCNT`. The documentation for `ComparatorFunction::CycleCount` already mentions the 0th comparator requirement

Reply:
The root issue in my eyes is requiring the user to pass the `DWT` to `configure` for an effectivly non-exhaustive check: while the current check ensures that `dwt[0]` supports cycle counters, `configure` still allows us to configure `dwt[1]` for it (effectively an invalid `configure` invocation, hardware-wise), as long as `dwt[0]` supports it. An exhaustive check would require two different types for `dwt[0]` and `dwt[n > 0]` and only allow cycle count comparisons for the `dwt[0]` type (as previously discussed), but that's an issue for another day.